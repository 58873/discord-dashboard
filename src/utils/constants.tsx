import { MenuGroup } from "./types";

export const menuItems: MenuGroup[] = [
  {
    name: "Security",
    path: "/security",
    routes: [
      {
        name: "Authentication",
        path: "/security/authentication",
      },
      {
        name: "Roles",
        path: "/security/roles",
      },
    ],
  },
  {
    name: "General",
    path: "/general",
    routes: [
      {
        name: "General",
        path: "/general/muted",
      },
      {
        name: "Roles",
        path: "/general/prefix",
      },
    ],
  },
  {
    name: "Miscellaneous",
    path: "/miscellaneous",
    routes: [
      {
        name: "Fun Commands",
        path: "/miscellaneous/fun-commands",
      },
      {
        name: "Levels",
        path: "/miscellaneous/levels",
      },
    ],
  },
];
