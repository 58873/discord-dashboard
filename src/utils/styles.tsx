import styled from "styled-components";

export const SidebarStyle = styled.aside`
  width: 100px;
  background-color: ${({ theme }) => theme.base.sidebar};
  color: ${({ theme }) => theme.text.main};
  height: 100%;
  font-family: "Poppins";
  box-sizing: border-box;
  position: fixed;
  overflow: scroll;
  ::-webkit-scrollbar {
    display: none;
  }
`;

export const MenuStyle = styled.aside`
  width: 350px;
  background-color: ${({ theme }) => theme.base.menu};
  color: ${({ theme }) => theme.text.main};
  height: 100%;
  font-family: "Poppins";
  box-sizing: border-box;
  position: fixed;
  margin-left: 100px;
  overflow: scroll;
  ::-webkit-scrollbar {
    display: none;
  }
`;

export const DashboardPageStyle = styled.div`
  height: 100%;
  background-color: ${({ theme }) => theme.base.content};
  color: ${({ theme }) => theme.text.main};
  font-family: "Poppins";
  box-sizing: border-box;
  position: absolute;
  margin-left: 450px;
  width: calc(100% - 450px);
`;

export const IconStyle = styled.div`
  height: 60px;
  width: 60px;
  background-color: #c4c4c4;
  border-radius: 50px;
`;

// Headers
export const BaseHeader = styled.header`
  box-sizing: border-box;
  height: ${({ theme }) => theme.height.topContent};
  box-shadow: 0px 1px 25px 0px rgba(0, 0, 0, 0.25);
  display: flex;
  align-items: center;
`;
export const SidebarHeader = styled(BaseHeader)`
  justify-content: center;
`;
export const DashboardHeader = styled(BaseHeader)`
  padding: 25px;
`;
export const MenuHeader = styled(BaseHeader)`
  padding: 25px;
  position: fixed;
  z-index: 9999;
  background-color: inherit;
  width: 100%;
  :hover {
    cursor: pointer;
  }
`;

export const SidebarContent = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: ${({ theme }) => theme.padding.content};
`;

export const GuildIcon = styled(IconStyle)`
  height: 60px;
  width: 60px;
  margin: 5px 0px;
`;

export const MenuContent = styled.div`
  box-sizing: border-box;
  padding: ${({ theme }) => theme.padding.content};
  margin-top: 100px;
`;

export const MenuCategory = styled.div`
  box-sizing: border-box;
  font-size: 16px;
  font-weight: bold;
  :nth-child(2) {
    margin: 20px 0px;
  }
  span {
    padding: 0px 20px;
  }
`;
export const MenuCategoryItem = styled.section`
  font-size: 22px;
  font-weight: lighter;
  padding: 5px 0px;
  transition: 100ms background-color;
  :hover {
    cursor: pointer;
    background-color: #42494c;
  }
  span {
    padding: 0px 20px;
  }
`;
