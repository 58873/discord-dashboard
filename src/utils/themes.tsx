export interface Theme {
  base?: any;
  text?: any;
  height?: any;
}

export const LightTheme: Theme = {};

export const DarkTheme: Theme = {
  base: {
    sidebar: "#202225",
    menu: "#2F3136",
    content: "#36393F",
  },
  text: {
    main: "#FFFFFF",
  },
  height: {
    topContent: "95px",
  },
};
