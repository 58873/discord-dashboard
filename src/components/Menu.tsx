import * as React from "react";
import {
  MenuStyle,
  MenuHeader,
  MenuContent,
  MenuCategory,
  MenuCategoryItem,
} from "../utils/styles";
import { menuItems } from "../utils/constants";
import { History } from "history";

type MenuProps = {
  title: string;
  history: History;
};

const mapCategories = (history: History) =>
  menuItems.map((item) => (
    <MenuCategory key={item.name}>
      <span>{item.name}</span>
      {item.routes.map((route) => (
        <MenuCategoryItem
          key={route.name}
          onClick={() => history.push(route.path)}
        >
          <span># {route.name}</span>
        </MenuCategoryItem>
      ))}
    </MenuCategory>
  ));

export const Menu = (props: MenuProps) => {
  console.log(props);
  return (
    <MenuStyle>
      <MenuHeader onClick={() => props.history.push("/dashboard")}>
        {props.title}
      </MenuHeader>
      <MenuContent>{mapCategories(props.history)}</MenuContent>
    </MenuStyle>
  );
};
