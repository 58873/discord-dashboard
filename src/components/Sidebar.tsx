import * as React from "react";
import { SidebarStyle } from "../utils/styles";

export const Sidebar = () => {
  return <SidebarStyle>asdsad</SidebarStyle>;
};
