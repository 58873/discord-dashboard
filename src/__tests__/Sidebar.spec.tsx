/* eslint-disable jest/no-mocks-import */
import * as React from "react";
import { render, screen } from "../utils/test";
import { Sidebar } from "../components/Sidebar";
import { guilds } from "../__mocks__";

it("should render Sidebar", () => {
  const { asFragment } = render(<Sidebar guilds={guilds} />);
  const firstRender = asFragment();
  expect(firstRender).toMatchSnapshot();
});

it("should render sidebar with 16 elements", () => {
  const { container } = render(<Sidebar guilds={guilds} />);
  const div = container.getElementsByClassName("guild-icon");
  expect(div.length).toBe(16);
});
