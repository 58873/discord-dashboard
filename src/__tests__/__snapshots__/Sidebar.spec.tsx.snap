// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sidebar 1`] = `
<DocumentFragment>
  <aside
    class="sc-bdfBwQ liFhOG"
  >
    <header
      class="sc-eCssSg sc-jSgupP eniBFN brmkFP"
    >
      <div
        class="sc-hKgILt eqmFsU"
      />
    </header>
    <div
      class="sc-fubCfw bjsDEv"
    >
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
      <div
        class="sc-hKgILt sc-pFZIQ eqmFsU fIPWgr guild-icon"
      />
    </div>
  </aside>
</DocumentFragment>
`;
